core/src/main/java/org/apache/calcite/avatica/remote/Driver.java

```diff
@@ -194,10 +196,20 @@ AvaticaHttpClient getHttpClient(AvaticaConnection connection, ConnectionConfig c
       } catch (Exception e) {
         long endTime = System.currentTimeMillis();
         long elapsedTime = endTime - startTime;
+        LOG.warn("Connection Failed: {}", e.getMessage());
         LOG.debug("Failure detected in: {} milliseconds", elapsedTime);
-        if (currentRetry < retries) {
+        if (currentRetry < retries - 1) {
```

Comment:
This means that if you set retries to 1, then you won't retry at all.
Technically, retry = 1 mens and initial attempt, and the one retry.
IMO the previous version was correct,

Reply:
The docs you wrote also says that this the number of REtries.

Reply:
Done.